src/main/java/com/onlinebookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+package com.onlinebookstore.service.impl;
+
+import com.onlinebookstore.dto.cart.CartItemUpdateDto;
+import com.onlinebookstore.dto.cart.CreateCartItemRequestDto;
+import com.onlinebookstore.dto.cart.ShoppingCartDto;
+import com.onlinebookstore.exception.EntityNotFoundException;
+import com.onlinebookstore.mapper.CartItemMapper;
+import com.onlinebookstore.mapper.ShoppingCartMapper;
+import com.onlinebookstore.model.Book;
+import com.onlinebookstore.model.CartItem;
+import com.onlinebookstore.model.ShoppingCart;
+import com.onlinebookstore.model.User;
+import com.onlinebookstore.repository.book.BookRepository;
+import com.onlinebookstore.repository.cart.CartItemRepository;
+import com.onlinebookstore.repository.cart.ShoppingCartRepository;
+import com.onlinebookstore.service.ShoppingCartService;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public void createShoppingCartForUser(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartDto getShoppingCartByUserId(Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't get shopping cart for user with id: " + userId));
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartDto addCartItem(Long userId, CreateCartItemRequestDto requestDto) {
+        Long bookId = requestDto.getBookId();
+        Book bookById = bookRepository.findById(bookId).orElseThrow(
+                () -> new EntityNotFoundException("Can't get book by id: " + bookId));
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId).orElseThrow(
+                () -> new EntityNotFoundException(
+                        "Can't get shopping cart for user with id: " + userId));
+        CartItem cartItem = cartItemMapper.toModel(requestDto);
+        cartItem.setBook(bookById);
+        cartItem.setShoppingCart(shoppingCart);
+        cartItemRepository.save(cartItem);
+        shoppingCart.getCartItems().add(cartItem);
+        shoppingCartRepository.save(shoppingCart);
+        return getShoppingCartByUserId(userId);
```

Comment:
What is the reason to call getShoppingCartByUserId here? Is it possible just return shopping cart?